### ISSUE TITLE
The MockPreparedStatement's batch is not cleared after executeBatch

### ISSUE THREAD

**Issue by kz282** [mockrunner/mockrunner#32]
Hello Carlos!

I have been testing batch inserts with Mockrunner-JDBC and encountered an issue: after I add parameter sets to a MockPreparedStatements batch via addBatch and then execute the batch via executeBatch the MockPreparedStatements batch is not cleared, so I must execute clearBatch manually before adding more parametersets to the batch. 

This question describes the issue properly: http://stackoverflow.com/questions/9192541/are-the-parameter-values-automatically-cleared-after-executing-a-batch-of-sql-co , and this answer links to the specification: http://stackoverflow.com/a/22005160

According to the 15.1.2 of the JDBC 3.0 specification I advise calling clearBatch() in the finally block of MockPreparedStatement like this:
```
--- origin/master (8cfeb9d)
+++ Modified In Working Tree
@@ -286,6 +286,8 @@
     
     protected int[] executeBatch(List<MockParameterMap> batchParams) throws SQLException
     {
+        try
+        {
         int[] results = new int[batchParams.size()];
         SQLException exception = null;
         for(int ii = 0; ii < results.length; ii++)
@@ -317,6 +319,11 @@
         }
         return results;
     }
+        finally
+        {
+            clearBatch();
+        }
+    }
 
     private void setGeneratedKeysResultSet(String sql, MockParameterMap params)
     {
```
Kind regards,
Zoltan

**Comment by kz282** [comment]
I have published a short patch with a test case for this issue here:
https://github.com/kz282/mockrunner/commit/c81d6d1b10371d6795c42937ce8b50754dba7389

Kind regards,
Zoltan

**Comment by cemartins** [comment]
Thanks Zoltan. This is very helpful.
I will run the tests with your code in place and, if it all passes, I can make another release with this fix.

Cheers
Carlos

**Comment by cemartins** [comment]
Hey Zoltan. I'm having a hard time copying the changes in your patch by hand. Could you make a pull request instead?

**Comment by cemartins** [comment]
Fix available in new release mockrunner-1.1.1
